src/main.rs

```diff
@@ -1,29 +1,69 @@
-use std::hash::Hash;
-use std::io::ErrorKind;
-use std::iter::FromIterator;
-use std::{
-    collections::HashSet,
-    fs::{self, File},
-    process::Command,
-};
-
+mod handler;
 use anyhow::{bail, Error, Result};
 use clap::{Parser, Subcommand, ValueEnum};
+use config::{Config, File, FileFormat};
 use glob::glob;
-use serde::{Deserialize, Serialize};
+use handler::*;
+use serde::Deserialize;
+use std::path::Path;
+use std::process::Command;
+use std::str;
+
+/// dir that greenboot looks for the health-check and other scripts
+static GREENBOOT_INSTALL_PATHS: [&str; 2] = ["/usr/lib/greenboot", "/etc/greenboot"];
+
+/// greenboot config path
+static GREENBOOT_CONFIG_FILE: &str = "/etc/greenboot/greenboot.conf";
 
 #[derive(Parser)]
 #[clap(author, version, about, long_about = None)]
 #[clap(propagate_version = true)]
+
+/// cli parameters for greenboot
 struct Cli {
     #[clap(value_enum, short, long, default_value_t = LogLevel::Info)]
     log_level: LogLevel,
-
     #[clap(subcommand)]
     command: Commands,
 }
+#[derive(Debug, Deserialize)]
+///config params for greenboot
+struct GreenbootConfig {
+    max_reboot: i32,
+}
+
+impl GreenbootConfig {
+    /// sets the default parameter for greenboot config
+    fn set_default() -> Self {
+        Self { max_reboot: 3 }
+    }
+    /// gets the config form the config file
+    fn get_config() -> Self {
+        let mut config = Self::set_default();
+        let parsed = Config::builder()
+            .add_source(File::new(GREENBOOT_CONFIG_FILE, FileFormat::Ini))
+            .build();
+        match parsed {
+            Ok(c) => {
+                config.max_reboot = match c.get_int("GREENBOOT_MAX_BOOT_ATTEMPTS") {
+                    Ok(c) => c.try_into().unwrap_or_else(|e| {
+                        log::warn!("{e}, using default value");
```

Comment:
Why are we using the default? Is it because the user hasn't defined it, or is it the defined value doesn't make sense. This message in the log will be confusing to users.

Reply:
It's not the e that bothers me here it's the string "using default value" after it. What does "using default value" in the context of this error. I think the string should read something around the fact it's using the max boot attempts defaults so that the user understands that else it's just a generic string in a log file with no other context.